code/scripting/api/libs/mission.cpp

```diff
@@ -955,6 +989,99 @@ ADE_FUNC(createWeapon,
 		return ade_set_error(L, "o", l_Weapon.Set(object_h()));
 }
 
+ADE_FUNC(createWarpeffect,
+	l_Mission,
+	"vector WorldPosition, vector PointTo, number radius, number duration /* Must be >= 4*/, fireballclass Class, "
+	"soundentry WarpOpenSound, soundentry WarpCloseSound, "
+	"[number WarpOpenDuration = -1, number WarpCloseDuration = -1, vector Velocity /* null vector by default*/, "
+	"boolean Use3DModel = false]",
+	"Creates a warp-effect fireball and returns a handle to it.",
+	"fireball",
+	"Fireball handle, or invalid fireball handle if fireball couldn't be created.")
+{
+	vec3d pos = vmd_zero_vector;
+	vec3d point_to = vmd_zero_vector;
+	float radius = 0.0f;
+	float duration = 4.0f;
+	int fireballclass = -1;
+	sound_entry_h *opensound = NULL;
+	sound_entry_h *closesound = NULL;
+
+	float opentime = -1.0f;
+	float closetime = -1.0f;
+	vec3d velocity = vmd_zero_vector;
+	bool warp3d = false;
+
+	if (!ade_get_args(L, "ooffooo|ffob", l_Vector.Get(&pos), l_Vector.Get(&point_to), &radius, &duration, l_Fireballclass.Get(&fireballclass), l_SoundEntry.GetPtr(&opensound), l_SoundEntry.GetPtr(&closesound), &opentime, &closetime, l_Vector.Get(&velocity), &warp3d)) {
+		return ade_set_error(L, "o", l_Fireball.Set(object_h()));
+	}
+
+	int flags = warp3d ? FBF_WARP_VIA_SEXP | FBF_WARP_3D : FBF_WARP_VIA_SEXP;
+
+	duration = duration < 4.0f ? 4.0f : duration;
+
+	// sanity check, if these were specified
+	if (duration < opentime + closetime)
+	{
+		//Both warp opening and warp closing must occur within the duration of the warp effect.
+		opentime = closetime = duration / 2.0f;
+	}
```

Comment:
For these kind of validation checks it would be good to cause a `LuaError` when the necessary condition is not met to allow the script writer to see that there is something wrong and where the error is coming from.

Reply:
Okay. Was thinking about this but wasn't sure if this was "worth" changing the behaviour of the function in comparison to it's sexp counterpart (that just does this).

Reply:
Even for SEXPs it is, in my opinion, a good idea to issue a warning about these things. Silently changing the values is likely to cause some frustration for the writer. For scripts it is even more useful since `LuaError` also prints a stacktrace of from where exactly the wrong function call came from.